File: backend/tests/services_api/retrieval/record.py
import aiohttp
from typing import Dict
from tests.common.utils import ResponseWrapper, get_headers, Token
from tests.settings import HOST
from app.config import CONFIG

if CONFIG.WEB:
    BASE_URL = f"{HOST}:{CONFIG.SERVICE_PORT}{CONFIG.WEB_ROUTE_PREFIX}"
elif CONFIG.API:
    BASE_URL = f"{HOST}:{CONFIG.SERVICE_PORT}{CONFIG.API_ROUTE_PREFIX}"
    from tests.common.utils import APIKEY

    Token = APIKEY


async def list_records(collection_id: str, params: Dict = None):
    headers = get_headers(Token)
    async with aiohttp.ClientSession(headers=headers) as session:
        request_url = f"{BASE_URL}/collections/{collection_id}/records"
        response = await session.get(request_url, params=params)
        return ResponseWrapper(response.status, await response.json())


async def get_record(collection_id: str, record_id: str):
    headers = get_headers(Token)
    async with aiohttp.ClientSession(headers=headers) as session:
        request_url = f"{BASE_URL}/collections/{collection_id}/records/{record_id}"
        response = await session.get(request_url)
        return ResponseWrapper(response.status, await response.json())


async def create_record(collection_id: str, payload: Dict):
    headers = get_headers(Token)
    async with aiohttp.ClientSession(headers=headers) as session:
        request_url = f"{BASE_URL}/collections/{collection_id}/records"
        response = await session.post(request_url, json=payload)
        return ResponseWrapper(response.status, await response.json())


async def update_record(collection_id: str, record_id: str, payload: Dict):
    headers = get_headers(Token)
    async with aiohttp.ClientSession(headers=headers) as session:
        request_url = f"{BASE_URL}/collections/{collection_id}/records/{record_id}"
        response = await session.post(request_url, json=payload)
        return ResponseWrapper(response.status, await response.json())


async def delete_record(collection_id: str, record_id: str):
    headers = get_headers(Token)
    async with aiohttp.ClientSession(headers=headers) as session:
        request_url = f"{BASE_URL}/collections/{collection_id}/records/{record_id}"
        response = await session.delete(request_url)
        return ResponseWrapper(response.status, await response.json())

File: frontend/src/axios/record.ts
import { request } from '../utils/index'

const getRecordsList = async <T extends Record<string,string>>(params:T,collectionId:string)=> {
const project_base_url = `api/v1`
    let str = ''
    if(params){
        Object.keys(params).forEach(key => {
            str+=`${key}=${params[key]}&`
         })
         str = str.substring(0,str.length-1)
    }
    return await request.get(`${project_base_url}/collections/${collectionId}/records?${str}`)
}
const createRecord = async (collectionId:string,params:object) => {
const project_base_url = `api/v1`
    return await request.post(`${project_base_url}/collections/${collectionId}/records`, params)
}
const deleteRecord = async (collectionId:string,id:string) => {
const project_base_url = `api/v1`
    return await request.delete(`${project_base_url}/collections/${collectionId}/records/${id}`)

}
const updateRecord = async (collectionId:string,id:string,params:object) => {
const project_base_url = `api/v1`
    return await request.post(`${project_base_url}/collections/${collectionId}/records/${id}`,params)
}
const getRecord = async (collectionId:string,id:string) => {
const project_base_url = `api/v1`
    return await request.get(`${project_base_url}/collections/${collectionId}/records/${id}`)
}
export { getRecordsList,createRecord,deleteRecord,updateRecord,getRecord }